### ISSUE TITLE
Add section on `tm_grid()`

### ISSUE THREAD

**Issue by prosoitos** [Robinlovelace/geocompr#391]
In section `8.2 Static maps`, I feel that a subsection on adding coordinate grid lines (with `tm_grid()`) is missing.

`ggplot2` includes such lines by default, which, in my opinion, is a sensible default. `tmap` does not. Knowing how to add them seems quite critical to me. Unless I have missed something (possible since I have still only read a rather small fraction of the book), there is nothing on this.

**Comment by prosoitos** [comment]
In addition, the `tmap` manual is everything but clear on how to use the `projection` argument of `tm_grid()`.

So some info on this in the book would be really useful.

**Comment by Robinlovelace** [comment]
I think that would be useful. Many thanks for spotting that and the suggestion!

**Comment by prosoitos** [comment]
I am still struggling to get a happy result with `tm_grid()` actually. ggplot2 produces nice, easy to read and pleasantly formatted coordinates while my results with `tm_grid()`, despite a lot of fiddling with various arguments are still pretty unattractive.

**Comment by prosoitos** [comment]
I guess, this could also be added into `8.2.5 Layouts` (rather than a new subsection). Which makes most sense probably depends on how much you want to write about it.

**Comment by Nowosad** [comment]
Hi @prosoitos, would you be able to prepare one or two examples of maps with coordinate grid lines using ggplot2? I plan to use it as a basis for developing a section of how to use `tm_grid()`.

**Comment by prosoitos** [comment]
ggplot2 has them by default. So as soon as you draw a map with it, it will have them (unless of course you play with the themes, options, etc.)

So if you throw any spacial object into

```r
ggplot() +
  geom_sf(data = <your-object>)
```

that is enough.

**Comment by prosoitos** [comment]
So, using one of the objects (`nz`) that you use quite a bit in the book, you could simply run:

```r
ggplot() + geom_sf(data = nz)
```

**Comment by prosoitos** [comment]
And if you want to customize them, you can add a `+ coord_sf()` function and play with its arguments (but I find the default satisfactory).

**Comment by prosoitos** [comment]
Compare for instance

```r
ggplot() + geom_sf(data = nz)
```

with

```r
tm_shape(nz) + tm_borders() + tm_grid()
```

I find the first pleasing and the 2nd nasty... So for someone who wants coordinate grids, I find tmap unfriendly. And if you give some way to customize `tm_grid()` nicely in the book, that could be very useful to many (I think).

**Comment by jannes-m** [comment]
Maybe related: https://github.com/mtennekes/tmap/issues/164

**Comment by prosoitos** [comment]
Yes, having axis ticks without gridlines would be really nice.

The unit and formatting of the coordinates themselves is also tricky to make bearable.

**Comment by jannes-m** [comment]
Check out also following code showing how to add tick marks with tmap: https://github.com/Robinlovelace/geocompr/blob/536af2801ef62ec93856d8c0a299461afbfdd4e7/11-spatial-cv.Rmd#L116

**Comment by prosoitos** [comment]
ggplot2 does not at all follow Tufte's principles and has horrible ink per area ratios. So not having the gray background but still having the coordinates in a ggplot2-style formats on ticks along the borders would be ideal.

**Comment by prosoitos** [comment]
> Check out also following code showing how to add tick marks with tmap:

Ah, I am not there yet in the book.

This is totally the kind of thing that I would expect to see in the `8.2.5 Layout` section.

**Comment by prosoitos** [comment]
I was wondering whether `tm_text()` should not be added in the layout section as well? Or at least mentioned.

**Comment by jannes-m** [comment]
Regarding the tick marks with **tmap**, I am not sure if I would present this probably rather inelegant hack in the book. Maybe it would be better to convince @mtennekes to add this functionality to **tmap** and/or asking him how we could support him in doing so.

**Comment by prosoitos** [comment]
I fully agree that `tmap` could improve there. I am too new to the package to open an issue on it, but I considered maybe doing so once I am more familiar with it (I want to make sure I am not missing already existing options).

However that is presented, I feel that things like `tm_text()` (which allow to have labels on polygons based on a variable), or some form of coordinates grid would be important to mention in the layout section. Chapter 8 contains a lot of information on colours and that is for sure critical in the aesthetics and readability of a map. But knowing how to change the color of the border for instance seems less critical to me than knowing how to add key elements in a plot. Adding a few more sections or adding to the layout section would, I think, re-balance things.

I hope not to be horribly annoying with my criticism though: I appreciate your work in putting this book together tremendously and only hope to improve it for future users!

**Comment by prosoitos** [comment]
Another thought I had, still on the theme of plot layout:

it is often necessary to define the boundaries of a map. How to do this with `st_bbox()` is explained in `8.2.7 Inset maps`. But it might be easier to find this information if that were explained earlier (in layout or in its own section), then referenced to in the inlet section: one often needs to zoom in an area of the data without planning on making inset maps.

**Comment by Robinlovelace** [comment]
A PR on this would be very welcome.

**Comment by Nowosad** [comment]
About `tm_grid()` - I think we should firstly try to improve the **tmap** defaults before trying to write a section with our hacks... Therefore, I opened an issue at https://github.com/mtennekes/tmap/issues/315.

**Comment by prosoitos** [comment]
Great! :slightly_smiling_face: Thanks @Nowosad. Great suggestions in that issue!

**Comment by prosoitos** [comment]
> A PR on this would be very welcome.

OK. I am quite busy in the coming few days, but happy to do this afterwards (only about defining boundaries. I won't do anything about `tm_grid()`).

**Comment by Robinlovelace** [comment]
And as Jakub says, it makes sense to document it only after https://github.com/mtennekes/tmap/issues/315 has been resolved. It's definitely an oversight in the book so thanks again for raising it and to Jakub to have the foresight and skills to propose upstream changes to **tmap** that everyone will benefit from.

**Comment by Nowosad** [comment]
Superseded by https://github.com/Robinlovelace/geocompr/issues/882
